### ISSUE TITLE
Parse error using jinja on version 22.1

### ISSUE THREAD

**Issue by shaneholder** [canonical/cloud-init#4490]
# Bug report
Hoping to get some guidance on Jinja templates with cloud-init 22.1 on RedHat 8.  I have a template that I'm putting in /etc/cloud/cloud.cfg.d. However when it runs I get a parse error, however if I remove the file from the cfg.d folder and execute it with --format it produces the correct output.  This behavior does not exhibit on Suse15 which has cloud-init 23.1 but only 22.1 is available in the RHEL repos.

## Steps to reproduce the problem
Put the following file in the cloud.cfg.d folder
```
## template: jinja
#cloud-config
runcmd:
{{ ds.meta_data.imds.compute.tagsList | selectattr('name', 'equalto', 'domain') | list }}
```
Executing with a template in cloud.cfg.d
```
sudo cloud-init query --format="$( cat /etc/cloud/cloud.cfg.d/02-resolv-conf.cfg )"
WARNING: Failed loading yaml blob. Invalid format at line 5 column 1: "while scanning a simple key
  in "<unicode string>", line 5, column 1:
    {{ ds.meta_data.imds.compute.tag ...
    ^
could not find expected ':'
  in "<unicode string>", line 6, column 1:
    
    ^"
## template: jinja
#cloud-config
runcmd:
[{'name': 'domain', 'value': 'my.domain'}]
```

but if I remove the file in cloud.cfg.d and execute it from my home folder it does parse correctly.

```
$ sudo rm /etc/cloud/cloud.cfg.d/02-resolv-conf.cfg
$ sudo cloud-init query --format="$( cat ./02-resolv-conf-tst.cfg )"
## template: jinja
#cloud-init
runcmd:
[{'name': 'domain', 'value': 'my.domain'}]
```

## Environment details
- Cloud-init version:  22.1
- Operating System Distribution: RedHat 8
- Cloud provider, platform or installer type: Azure

## cloud-init logs

[cloud-init.tar.gz](https://github.com/canonical/cloud-init/files/12821661/cloud-init.tar.gz)

**Comment by TheRealFalcon** [comment]
@shaneholder , unfortunately Jijja templating for files in /etc/cloud/* isn't available before 22.4, so it will not work on 22.1. It was added via https://github.com/canonical/cloud-init/pull/1722

Since this is fixed in more recent versions of cloud-init, I'm going to close this issue. Feel free to re-open if I have missed something or you believe it hasn't been addressed.

**Comment by shaneholder** [comment]
Makes perfect sense thank you for the quick response.  At least now I'm not wondering what I'm doing wrong.
